action.yml

```
@@ -8,6 +8,9 @@ inputs:
     description: 'Path to config (relative to $GITHUB_WORKSPACE)'
     required: false
     default: '.github/.gitleaks.toml'
+  branch:
+    description: 'Branch to scan'
+    required: true
```

Would make sense to have a sensible default with main?
```suggestion
    required: false
    default: 'main'
```

---

I thought about it. But as the community is also using the current version. I didn't want to disrupt the behavior which is:
on PRs, if the workflow is actionable on the push event also, then gitleaks scans the entire history on all existing branches during the push run.

At least that's what I observed which led me to investigate on the problem and find out that we could inject the current branch.

If we pass a default, then I don't know how people could reproduce the current behavior, but this would require an update of their step.

---

But there's no doubt the required should be false.

---

> @RUBenGAMArrarodRiguEZ-ToMtOm
> 
> I thought about it. But as the community is also using the current version. I didn't want to disrupt the behavior which is: on PRs, if the workflow is actionable on the push event also, then gitleaks scans the entire history on all existing branches during the push run.
> 
> At least that's what I observed which led me to investigate on the problem and find out that we could inject the current branch.
> 
> If we pass a default, then I don't know how people could reproduce the current behavior, but this would require an update of their step.

Thanks for the clarification. From what I understood, the previous behavior was:
For every push, no matter which branches were pushed, scan all the branches.

And adding `main` as default would have been a breaking change. Correct?